### PR TITLE
[FIX]  Flextab information is not working when clicking on visitor or agent username in Omnichannel messages

### LINKED ISSUE #16795: Visitor and livechat agent info flexTab data is not defined if accessed by clicking guest/agent name/avatar
### Description:
When a livechat agent clicks on guest-user or livechat-agent's name or icon then blank user-info flexTabBar is rendered.

### Steps to reproduce:

1. Chat with a guest as livechat agent
2. Click on username or avatar of livechat agent or guest

### Expected behavior:
It should open the user-info same way as it is opened by clicking user-info button on the room header

### Actual behavior:
It is opened blank with no data.

### PR DESCRIPTION
**Pull request by ashwaniYDV**
Closes #16795

### CODE REVIEW

**Comment by ashwaniYDV** [comment]
@renatobecker Please guide me better aproach to check if a user is livechat guest.
I have temporarily used this regex `if (/^guest-(\d+)$/.test(username))`

**Comment by ashwaniYDV** [comment]
@renatobecker Please review :)

File: app/ui/client/views/app/room.js [hunk 1]
```diff
@@ -799,6 +799,29 @@ Template.room.events({
 		}
 
 		const { username } = msg.u;
+		const roomData = Session.get(`roomData${ RoomManager.openedRoom }`);
```
Comment (renatobecker) [inline app/ui/client/views/app/room.js]: I don't think this is a better implementation of this issue.
My suggestion is to create a custom method on the [roomTypes](https://github.com/RocketChat/Rocket.Chat/blob/develop/app/utils/client/lib/roomTypes.js#L8) class, called by something like `openCustomProfileTab`. In the default method, you would return false and in the `LivechatRoomType` you would implement the method, passing the instance, roomData and username as parameters. Then, you just need to check if the username is the same as the guest username, if so you could do something we do [here](https://github.com/RocketChat/Rocket.Chat/blob/develop/app/livechat/client/hooks/onRenderRoom.js#L12);

**Comment by ashwaniYDV** [comment]
hi @renatobecker. I have done the requested changes

File: app/ui/client/views/app/room.js [hunk 2]
```diff
@@ -63,9 +63,11 @@ const openProfileTab = (e, instance, username) => {
 		instance.userDetail.set(username);
 	}
 
-	instance.groupDetail.set(null);
-	instance.tabBar.setTemplate('membersList');
-	instance.tabBar.open();
+	if (!roomTypes.roomTypes[roomData.t].openCustomProfileTab(instance, roomData, username)) {
```
Comment (renatobecker) [inline app/ui/client/views/app/room.js]:
```suggestion
	if (roomTypes.roomTypes[roomData.t].openCustomProfileTab(instance, roomData, username)) {
	    return;
	}
```

**Pull request merged**